src/Equinox.Core/Internal.fs

```diff
@@ -11,3 +11,35 @@ module Log =
     let [<return: Struct>] (|ScalarValue|_|) : Serilog.Events.LogEventPropertyValue -> obj voption = function
         | :? Serilog.Events.ScalarValue as x -> ValueSome x.Value
         | _ -> ValueNone
+
+module TaskSeq =
+
+    open FSharp.Control
+
+    let takeWhile predicate (source: taskSeq<_>) = taskSeq {
+        use e = source.GetAsyncEnumerator(System.Threading.CancellationToken())
+        let! step = e.MoveNextAsync()
+        let mutable go = step
+        while go do
+            let value = e.Current
+            if predicate value then
+                yield value
+                let! more = e.MoveNextAsync()
+                go <- more
+            else
+                go <- false
+    }
```

Comment:
Might be worth contributing back to taskSeq?

Their docs seem to suggest that it's intended to be there https://github.com/fsprojects/FSharp.Control.TaskSeq#progress-taskseq-ce